src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java

```diff
@@ -106,6 +112,91 @@ public ProfessionalUser findProfessionalUserByEmailAddress(String email) {
+            }
+            return findRefreshUsers(since, searchAfter);
+        }
+        return findSingleRefreshUser(userId);
+    }
+
+    public ResponseEntity<Object> findRefreshUsers(String since, UUID searchAfter) {
+        LocalDateTime formattedSince = LocalDateTime.parse(since, ISO_DATE_TIME_FORMATTER);
+
+        List<ProfessionalUser> professionalUsers;
+
+        if (searchAfter == null) {
+            professionalUsers = professionalUserRepository.findByLastUpdatedGreaterThanEqual(
+                    formattedSince
+            );
+        } else {
+            professionalUsers = professionalUserRepository.findByLastUpdatedGreaterThanEqualAndIdGreaterThan(
+                    formattedSince, searchAfter
+            );
+        }
+
+        List<UserConfiguredAccess> userConfiguredAccesses = professionalUsers.stream()
+                .map(ProfessionalUser::getUserConfiguredAccesses)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+
+        GetRefreshUsersResponse res = RefDataUtil.buildGetRefreshUsersResponse(
+                null, professionalUsers, userConfiguredAccesses
+        );
+
+        return ResponseEntity.status(HttpStatus.OK).body(res);
+    }
+
+    public ResponseEntity<Object> findRefreshUsersPageable(String since, Pageable pageable, UUID searchAfter) {
+        LocalDateTime formattedSince = LocalDateTime.parse(since, ISO_DATE_TIME_FORMATTER);
+
+        Page<ProfessionalUser> professionalUsersPage;
+
+        if (searchAfter == null) {
+            professionalUsersPage = professionalUserRepository.findByLastUpdatedGreaterThanEqual(
+                    formattedSince, pageable
+            );
+        } else {
+            professionalUsersPage = professionalUserRepository.findByLastUpdatedGreaterThanEqualAndIdGreaterThan(
+                    formattedSince, searchAfter, pageable
+            );
+        }
+
+        List<ProfessionalUser> professionalUsers = professionalUsersPage.getContent();
+
+        List<UserConfiguredAccess> userConfiguredAccesses = professionalUsers.stream()
+                .map(ProfessionalUser::getUserConfiguredAccesses)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+
+        GetRefreshUsersResponse res = RefDataUtil.buildGetRefreshUsersResponse(
+                professionalUsersPage, professionalUsers, userConfiguredAccesses
+        );
+
+        return ResponseEntity.status(HttpStatus.OK).body(res);
+    }
+
+    public ResponseEntity<Object> findSingleRefreshUser(String userId) {
+        ProfessionalUser professionalUser = professionalUserRepository.findByUserIdentifier(userId);
+        List<UserConfiguredAccess> userConfiguredAccesses;
+
+        if (professionalUser != null) {
+            userConfiguredAccesses = professionalUser.getUserConfiguredAccesses();
+        } else {
+            throw new ResourceNotFoundException("User does not exist");

```

Comment:
Should this be a Status return of either 403 or 404 as user does not exist?

Reply:
yes - by throwing `ResourceNotFoundException` a 404 is thrown. see https://github.com/hmcts/rd-professional-api/blob/master/src/main/java/uk/gov/hmcts/reform/professionalapi/controller/advice/ExceptionMapper.java#L67 for code that handle exception & return status code 404